sqlglot/dialects/presto.py

```diff
@@ -188,10 +188,16 @@ def _jsonextract_sql(self: Presto.Generator, expression: exp.JSONExtract) -> str
         )
 
     this = self.sql(expression, "this")
-    expr = self.sql(expression, "expression")
 
-    # Convert the JSONPath extraction `JSON_EXTRACT(col, '$.k1.k2') to a ROW access col.k1.k2
-    expr = expr.replace("'", "").lstrip("$")
+    # Convert the JSONPath extraction `JSON_EXTRACT(col, '$.x.y) to a ROW access col.x.y
+    segments = []
+    for path_key in expression.find_all(exp.JSONPathKey):
```

Comment:
why are we doing `find_all` here? looks like a smell, this doesn't guarantee an order

Reply:
Thought that BFS should be fine but swapped to a for-loop to make sure